### PR TITLE
Add string diff to the error message when equality assertion fails

### PR DESCRIPTION
**Pull request by Dilhasha**
## Purpose
Adds string diff to the error message when equality assertion fails.

Fixes part 2 of #12892

## Approach
 
Note: Add "java-diff-utils-4.5.jar"(54KB) to "ballerina-lang/distribution/zip/jballerina-tools/lib/"

When "assertEquals" fails and the types of the compared values are the same, a string difference will be shown as in the below samples.

## Samples

```
> test:assertEquals("hello dilhassha", "hello dilhasha");

            Assertion Failed!
                expected: 'hello dilhasha'
                actual  : 'hello dilhassha'
                Diff    :
                --- expected
                +++ actual
                @@ -1,1 +1,1 @@
                -hello dilhassha
                +hello dilhasha
```

 ```
>
string value1 = "Ballerina is an open source programming language and platform for cloud-era application programmers.\nSequence diagrams have been everyone’s favorite tool to describe how distributed & conccurrent programs work.";
string value2 = "Ballerina is an open source programming language and platform for cloud-era application programmersss.\nSequence diagrams have been everyone’s favorite tool to describe how distributed & concurrent programs work.";
test:assertEquals(value1,value2);

        Assertion Failed!
                expected: 'Ballerina is an open source programming language and platform for cloud-era appl'
                actual  : 'Ballerina is an open source programming language and platform for cloud-era appl'
                Diff    :
                --- expected
                +++ actual
                @@ -1,4 +1,4 @@
                Ballerina is an open source programming language and platform for cloud-era appl
                -ication programmers.
                +ication programmersss.
                Sequence diagrams have been everyone’s favorite tool to describe how distributed
                - & conccurrent programs work.
                + & concurrent programs work.
```

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by Dilhasha** [comment]
Assertion diff for other data types will be addressed in https://github.com/ballerina-platform/ballerina-lang/issues/26204

File: misc/testerina/modules/testerina-core/src/main/java/module-info.java [hunk 1]
```diff
@@ -4,4 +4,5 @@
     requires io.ballerina.core;
     requires io.ballerina.testerina.runtime;
     requires io.ballerina.jvm;
+    requires io.github.javadiffutils;
 }
```
Comment (parkavi11) [inline misc/testerina/modules/testerina-core/src/main/java/module-info.java:8]: End of line is missing

File: misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/AssertionDiffEvaluator.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
+package org.ballerinalang.testerina.core;
```
Comment (parkavi11) [inline misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/AssertionDiffEvaluator.java:1]: Why don't we add license header for this file?

File: tests/testerina-integration-test/src/test/resources/project-based-tests/basic-tests/src/assertions-error-messages/tests/assertions-error-messages.bal [hunk 3]
```diff
@@ -88,7 +88,7 @@ function testAssertTableAndString() {
     string customerTabString = "table [{id: 1, name: \"John\", salary: 300.50},{id: 2, name: \"Bella\", salary: 500.50}]";
     error? err = trap test:assertEquals(customerTab, customerTabString);
     error result = <error>err;
-    test:assertEquals(result.message().toString(), "Assertion Failed!\nexpected: <string> 'table [{id: 1, name: \"John\", salary: 300.50},{id: 2, name: \"Bella\", salary: 500.'\nactual\t: <table> '[{\"id\":1,\"name\":\"John\",\"salary\":300.5},{\"id\":2,\"name\":\"Bella\",\"salary\":500.5}]'");
+    test:assertEquals(result.message().toString(), "Assertion Failed!\nexpected: <string> 'table [{id: 1, name: \"John\", salary: 300.50},{id: 2, name: \"Bella\", salary: 500....'\nactual\t: <table> '[{\"id\":1,\"name\":\"John\",\"salary\":300.5},{\"id\":2,\"name\":\"Bella\",\"salary\":500.5}]'");
```
Comment (parkavi11) [inline tests/testerina-integration-test/src/test/resources/project-based-tests/basic-tests/src/assertions-error-messages/tests/assertions-error-messages.bal:91]: Isn't this line exceeds 120 character limit?

**Review by hevayo — APPROVED** [review]

**Pull request merged**